### PR TITLE
Add goldfish timer driver support

### PR DESCRIPTION
**Pull request by GUIDINGLI**
## Summary

Add goldfish timer driver support

## Impact

new driver, none

## Testing

Goldfish

### CODE REVIEW

**Comment by nuttxpr** [comment]
[**\[Experimental Bot, please feedback here\]**](https://github.com/search?q=repo%3Aapache%2Fnuttx+13552&type=issues)



**No, this PR does not meet the NuttX Requirements.**

**Here's why and how to improve it:**

* **Summary:**
    *  **Insufficient Detail:** "Add goldfish timer driver support" is too vague.
        * **What is the "goldfish timer"?** Is it a hardware timer on a specific platform? A simulated timer? 
        * **How does this driver work?** What timer API does it use?  

* **Impact:**
    * **Too Brief:** "new driver, none" needs expansion. 
        * **Be specific about potential impact:**  Even if you believe there is no user impact, state it explicitly (e.g., "No changes to user-facing APIs").
        * **Consider all categories:** Address *all* impact categories, even if the answer is "NO".  This demonstrates you've thought about potential side-effects.

* **Testing:**
    * **Missing Information:**
        * **Build Host:**  What operating system, architecture, and compiler did you use to build NuttX?
        * **"Goldfish" is not enough detail:** Which specific Goldfish platform/emulator? What configuration?
    * **No Logs:**  You need to provide actual testing logs *before* and *after* your changes to show the problem you're solving and that your solution works.

**Example of an Improved Summary:**

> This PR adds a new timer driver for the QEMU Goldfish platform's emulated timer. The driver leverages the existing NuttX timer framework and exposes a system timer for OS scheduling. It interfaces directly with the Goldfish platform's timer registers through memory-mapped I/O.

**To make your PR ready for review:**

1. **Expand the Summary:** Provide a clear and detailed explanation of the change, its purpose, and how it works.
2. **Complete the Impact Section:** Address *all* impact categories with specific details, even if the answer is "NO."
3. **Provide Complete Testing Information:** Include build host details, specific target platform/configuration, and relevant *before* and *after* testing logs.

File: drivers/timers/Kconfig [hunk 1]
```diff
@@ -515,4 +515,9 @@ config CS2100CP_REGDEBUG
 	depends on DEBUG_FEATURES
 
 endif # TIMERS_CS2100CP
+
+config GOLDFISH_TIMER
+	bool "Enable GOLDFISH_TIMER"
+	default n
```
Comment (acassis) [inline drivers/timers/Kconfig:521]: Please include the ---help--- to give more information to the end users

**Review by acassis — APPROVED** [review]

**Comment by acassis** [comment]
@GUIDINGLI as the bot noticed there are many missing information, also the Documentation/ about Goldfish is null:
https://nuttx.apache.org/docs/latest/platforms/arm/goldfish/index.html

**Comment by acassis** [comment]
@GUIDINGLI please add some basic Documentation/ about the Goldfish board emulation

**Comment by GUIDINGLI** [comment]
> @GUIDINGLI please add some basic Documentation/ about the Goldfish board emulation

Docs are updated

**Comment by GUIDINGLI** [comment]
@acassis

**Review by xiaoxiang781216 — APPROVED** [review]

**Pull request merged**